ddl/column.go

```diff
@@ -573,10 +573,21 @@ func checkDropColumnForStatePublic(tblInfo *model.TableInfo, colInfo *model.Colu
 }
 
 func onDropColumn(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
-	tblInfo, colInfo, idxInfos, err := checkDropColumn(t, job)
+	tblInfo, colInfo, idxInfos, ifExists, err := checkDropColumn(t, job)
 	if err != nil {
+		if ifExists && dbterror.ErrCantDropFieldOrKey.Equal(err) {
```

Comment:
Please add a comment for it. Why need to check ErrCantDropFieldOrKey, is it enough?

Reply:
It is used to ignore the `ifExists` warning. If the column exists, `ErrCanDropFieldOrKey` is the only error can be reported. For details, see `checkDropColumn`.